src/basic/path-lookup.c

```diff
@@ -91,6 +91,33 @@ int xdg_user_data_dir(char **ret, const char *suffix) {
         return 1;
 }
 
+int runtime_directory(char **ret, const char *suffix) {
+        int r;
+        _cleanup_free_ char *d = NULL;
+
+        assert(ret);
+        assert(suffix);
+
+        const char *e = getenv("RUNTIME_DIRECTORY");
+        if (e)
+                d = strdup(e);
+        else if (getuid() == 0)
+                d = strdup("/run");
```

Comment:
this isn't right. $RUNTIME_DIRECTORY is generally already suffixed, but /run obviously isn't.